scrape/scrape.go

```diff
@@ -1569,18 +1572,39 @@ loop:
 		// Increment added even if there's an error so we correctly report the
 		// number of samples remaining after relabeling.
 		added++
-
+		exemplars = exemplars[:0] // Reset and reuse the exemplar queue.
 		for hasExemplar := p.Exemplar(&e); hasExemplar; hasExemplar = p.Exemplar(&e) {
 			if !e.HasTs {
 				e.Ts = t
 			}
+			exemplars = append(exemplars, e)
+			e = exemplar.Exemplar{} // Reset for next time round loop.
+		}
+		sort.Slice(exemplars, func(i, j int) bool {
+			if exemplars[i].Ts != exemplars[j].Ts {
+				return exemplars[i].Ts < exemplars[j].Ts
+			}
+			return exemplars[i].Value < exemplars[j].Value
```

Comment:
I guess two exemplars could have the same timestamp, the same value, _but_ different labels. Which would mean they are different. How about this:

```suggestion
			if exemplars[i].Value != exemplars[j].Value {
				return exemplars[i].Value < exemplars[j].Value
			}
			return exemplars[i].Labels.Hash() < exemplars[j].Labels.Hash()
```